File: python-cli/markdown_test.py
#!/usr/bin/env python3
"""
Rich Python Markdown Test CLI
Reads and renders a markdown file
Run: python python-cli/markdown_test.py
"""
from rich.console import Console
from rich.markdown import Markdown
from pathlib import Path

console = Console()

def main():
    markdown_file_path = '/Users/leemoore/code/rich-port/rich/rich-ts/phases/COMPLETE_PORT_PLAN.md'
    
    try:
        console.print('[bold cyan]Reading markdown file:[/bold cyan]', markdown_file_path)
        console.print('')
        
        # Read the markdown file
        with open(markdown_file_path, 'r', encoding='utf-8') as f:
            markdown_content = f.read()
        
        # Render it with Rich
        markdown = Markdown(markdown_content)
        console.print(markdown)
        
        console.print('')
        console.print('[green]✓[/green] Markdown rendered successfully!')
    except FileNotFoundError:
        console.print(f'[red]✗[/red] File not found: {markdown_file_path}')
        exit(1)
    except Exception as error:
        console.print('[red]✗[/red] Error reading or rendering markdown:')
        console.print(f'[red]{error}[/red]')
        import traceback
        console.print(f'[dim]{traceback.format_exc()}[/dim]')
        exit(1)

if __name__ == '__main__':
    main()

File: typescript-cli/markdown-test.ts
#!/usr/bin/env node
/**
 * Rich TypeScript Markdown Test CLI
 * Reads and renders a markdown file
 * Run: npm run markdown-test
 */
import { Console } from '../rich-ts/src/console.js';
import { Markdown } from '../rich-ts/src/markdown.js';
import { readFileSync } from 'fs';

const console = new Console({
  force_terminal: true,
  highlight: true,
  markup: true,
});

async function main() {
  const markdownFilePath = '/Users/leemoore/code/rich-port/rich/rich-ts/phases/COMPLETE_PORT_PLAN.md';
  
  try {
    console.print('[bold cyan]Reading markdown file:[/bold cyan]', markdownFilePath);
    console.print('');
    
    // Read the markdown file
    const markdownContent = readFileSync(markdownFilePath, 'utf-8');
    
    // Render it with Rich
    const markdown = new Markdown(markdownContent);
    console.print(markdown);
    
    console.print('');
    console.print('[green]✓[/green] Markdown rendered successfully!');
  } catch (error) {
    console.print('[red]✗[/red] Error reading or rendering markdown:');
    console.print(`[red]${error instanceof Error ? error.message : String(error)}[/red]`);
    if (error instanceof Error && error.stack) {
      console.print(`[dim]${error.stack}[/dim]`);
    }
    process.exit(1);
  }
}

main().catch((error) => {
  console.print(`[red]Fatal error:[/red] ${error instanceof Error ? error.message : String(error)}`);
  if (error instanceof Error && error.stack) {
    console.print(`[dim]${error.stack}[/dim]`);
  }
  process.exit(1);
});

